src/prefect/cli/deployment.py

```diff
@@ -933,7 +934,20 @@ async def delete(
         $ prefect deployment delete --id dfd3e220-a130-4149-9af6-8d487e02fea6
     """
     async with get_client() as client:
-        if name is None and deployment_id is not None:
+        if _all:
```

Comment:
could be worth while to exit with error if `name` or `id` are provided along with `--all`